src/components/ItemList/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect } from 'react';
+import { updateItemDate } from '../Utils/firestore';
+import { differenceInHours } from 'date-fns';
+
+function ItemList({ itemName, docId, lastPurchasedDate }) {
+  const [isChecked, setIsChecked] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [token] = useState(() => window.localStorage.getItem('tcl18-token'));
+  const toggle = () => setIsChecked(!isChecked);
+
+  const editProduct = (id) => {
+    updateItemDate(token, id)
+      .then(() => {
+        console.log('product updated');
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  useEffect(() => {
+    if (isChecked) {
+      updateItemDate(token, docId).then(() =>
+        console.log('The date was updated'),
+      );
+    }
+  }, [isChecked, docId, token]);
+
+  useEffect(() => {
+    if (lastPurchasedDate && isDisabled === false) {
+      const hoursAtTheMoment = differenceInHours(
+        new Date(),
+        lastPurchasedDate.toDate(),
+      );
+      setIsDisabled(hoursAtTheMoment <= 12);
```

Comment:
I'm not sure but I believe that you need to compare with `<=24` because you get some difference between two dates in hours (or maybe it's much better with `differenceInSeconds` to reach some more precision (1hr = 86400 sec), it's only suggestions 😄  )

Reply:
That part of code was refactored.